basic-services/README.md

```diff
@@ -0,0 +1,176 @@
+or as pre-release versions of 1.0 (follow microsoft guidance here on prerelease versioning: https://docs.microsoft.com/en-us/nuget/create-packages/prerelease-packages )
+    Breaking changes should align with major versions of Metasys where possible. We should avoid any breaking changes
+    in between release of Metasys.
+    
+* Naming guidelines: Follow Microsoft Naming guidelines https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/naming-guidelines 
+
+* All submitted code must be reviewed through pull requests.
+
+* Good commit messages must be used on all commits. https://chris.beams.io/posts/git-commit/ 
+Most commits should NOT be one liners. When adding a commit add a paragraph or a few paragraphs explaining what you did.
+Only the smallest of commits should be one line messages.
+
+* Consider adding a light weight CONTRIBUTING.md or referencing Microsoft's for guidance for outsiders and others in JCI.
+
+* Track the Issues on this repo and provide guidance (where possible) on roadmap.
+
+### Phase 1
+
+We will begin on what I'm calling the TraditionalClient. (name can be changed).
+I'm calling it the TraditionalClient because of its heavy influence on the old MSSDA.
+The traditional client is probably ideal for the typical MSSDA customer because it 
+doesn't utilize many advanced C# features. But it is non-ideal for advanced development.
+
+1. It is synchronous. A more advanced SDK would expose all operations as asynch operations.
+2. It doesn't use generics. Instead it returns simple return structures (like ReadPropertyResult) that
+   model attribute values as strings, numbers and array. The client just needs to know based on what
+   attribute they are reading which of the values is most appropriate for thier usage. This is how 
+   MSSDA worked as well. A more advanced SDK would use `JToken` values or `dynamic` or generics.
+3. It doesn't use Nullable<T> which makes it a little awkward in some cases. Like what is the numeric value
+   of the string "AV1"? Using the MSSDA guidelines the numeric value of a string will be 0. (A more sophisticated
+   API would mark the numeric value as nullable for when it doesn't make sense)
+   
+The following requirements will be met in phase 1:
+
+1. Implement Login
+1. Implement GetObjectIdentifier
+1. Implement ReadProperty
+2. Implement ReadPropertyMultiple
+3. Expose the library as COM dll for use in Excel apps
+4. Ensure signatures of the methods are easy to consume via COM
+5. Must provide client side resource files for translating enum values
+   coming back from the server. (Unlike MSSDA, enums are not returned
+   from the server as localized strings. They are returned as "symbolic
+   strings" which are more useful at the API level. The client side
+   resource files will provide the ability for clients to show localized
+   strings.
+6. Nuget package support (for deployment to metasys-server org)
```

Comment:
Might be a dumb question, but are there other repos that we should be using instead of nuget for the customers needs.  NPM?  Other?  I'm not as familiar with public repos so I may be asking a non-sensical question.

Reply:
Yes, npm is where we'll put our JavaScript client when developed. Nuget is the appropriate location for dotnet packages.